pkg/storage/storage.go

```diff
@@ -174,3 +228,308 @@ func (s *svc) DeleteUploadID(ctx context.Context, user, bucket, object, uploadID
 	val := fmt.Sprintf("%s:%s", object, uploadID)
 	return s.client.SRem(ctx, key, val).Err()
 }
+
+func (s *svc) GetLastListedConsistencyCheckObj(ctx context.Context, obj *ConsistencyCheckObject) (string, error) {
+	key := fmt.Sprintf("ccv:l:%s:%s:%s", obj.ConsistencyCheckID, obj.Storage, obj.Prefix)
+	cmd := s.client.Get(ctx, key)
+	err := cmd.Err()
+
+	switch {
+	case errors.Is(err, redis.Nil):
+		return "", nil
+	case err != nil:
+		return "", fmt.Errorf("unable to set last listed object: %w", err)
+	default:
+		return cmd.Val(), nil
+	}
+}
+
+func (s *svc) SetLastListedConsistencyCheckObj(ctx context.Context, obj *ConsistencyCheckObject, value string) error {
+	key := fmt.Sprintf("ccv:l:%s:%s:%s", obj.ConsistencyCheckID, obj.Storage, obj.Prefix)
+	if err := s.client.Set(ctx, key, value, lastListedObjTTL).Err(); err != nil {
+		return fmt.Errorf("unable to set last listed object: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) DeleteLastListedConsistencyCheckObj(ctx context.Context, obj *ConsistencyCheckObject) error {
+	key := fmt.Sprintf("ccv:l:%s:%s:%s", obj.ConsistencyCheckID, obj.Storage, obj.Prefix)
+	if err := s.client.Unlink(ctx, key).Err(); err != nil {
+		return fmt.Errorf("unable to delete last listed object: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) DeleteAllLastListedConsistencyCheckObj(ctx context.Context, id string) error {
+	keyPattern := fmt.Sprintf("ccv:l:%s:*", id)
+	var cursor uint64
+
+	for {
+		keys, nextCursor, err := s.client.Scan(ctx, cursor, keyPattern, 0).Result()
+		if err != nil {
+			return fmt.Errorf("unable to scan keys: %w", err)
+		}
+
+		for _, key := range keys {
+			s.client.Unlink(ctx, key)
```

Comment:
how many commands are expected here? if there is an entry per object, then it will be better to use `redis.Pipeline` to batch commands.

Reply:
Added pipeline.